pyccel/ast/functionalexpr.py

```diff
@@ -59,24 +57,27 @@ def index(self):
         return self._index
 
 #==============================================================================
-class GeneratorComprehension(AtomicExpr, Basic):
-    _attribute_nodes = ()
+class GeneratorComprehension(FunctionalFor):
+    """ Super class for all functions which reduce generator expressions to scalars
+    """
 
 #==============================================================================
-class FunctionalSum(GeneratorComprehension, FunctionalFor):
-    _attribute_nodes = FunctionalFor._attribute_nodes
+class FunctionalSum(GeneratorComprehension):
+    """ Represents a call to sum for a list argument
+    >>> sum([i in range(5)])
+    """
     name = 'sum'
 
 #==============================================================================
-class FunctionalMax(GeneratorComprehension, FunctionalFor):
-    _attribute_nodes = FunctionalFor._attribute_nodes
+class FunctionalMax(GeneratorComprehension):
+    """ Represents a call to max for a list argument
+    >>> max([i in range(5)])
+    """
     name = 'max'
 #==============================================================================
 
-class FunctionalMin(GeneratorComprehension, FunctionalFor):
-    _attribute_nodes = FunctionalFor._attribute_nodes
+class FunctionalMin(GeneratorComprehension):
+    """ Represents a call to min for a list argument
+    >>> min([i in range(5)])
+    """
     name = 'min'
-
-#==============================================================================
-class FunctionalMap(GeneratorComprehension, FunctionalFor):
-    _attribute_nodes = FunctionalFor._attribute_nodes
```

Comment:
Did you remove this class intentionally?

Reply:
It is not used and we have the class `PythonMap`. If there is a difference between the two classes then we should keep this one but I realised I couldn't think of one when I tried to add the docstring